pkg/kubelet/cm/devicemanager/endpoint.go

```diff
@@ -182,6 +190,15 @@ func (e *endpointImpl) allocate(devs []string) (*pluginapi.AllocateResponse, err
 	})
 }
 
+// initialize issues Initialize gRPC call to the device plugin.
+func (e *endpointImpl) initialize(devs []string) (*pluginapi.Empty, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
```

Comment:
I also think you should store the context (and cancel function) in the endpoint object.

The next PR is going to be about hooking into the Kubelet pod remove code path so that we can cancel the Init call (and probably the Allocate too, but that's a different story).

Reply:
I would prefer to do this with next PR.